mrp_project/models/mrp_production.py

```diff
@@ -91,3 +91,8 @@ class MrpProductionWorkcenterLine(models.Model):
     work_ids = fields.One2many(
         comodel_name="project.task.work", inverse_name="workorder",
         string="Task works")
+
+    @api.multi
+    def write(self, vals, update=True):
+        return super(MrpProductionWorkcenterLine, self.with_context(
+            production_id=self.production_id.id)).write(vals)
```

Comment:
It's better practice pass the full recordset instead only the ID, in case you need any field for not making the browse again.